src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseResource.java

```diff
@@ -139,7 +143,41 @@ else if (authCheckService.isAtLeastTeachingAssistantForExercise(exercise, user))
     }
 
     /**
-     * GET /exercises/:exerciseId : get the "exerciseId" exercise with data useful for tutors.
+     * GET /exercises/:exerciseId/example-solution : get the exercise with example solution without sensitive fields
+     * if the user is allowed to access the example solution and if the example solution is published.
+     *
+     * @param exerciseId the exerciseId of the exercise with the example solution
+     * @return the ResponseEntity with status 200 (OK) and with body the exercise with its example solution after filtering sensitive data, or with
+     * status 404 (Not Found) is the exercise is not found, or with status 403 (Forbidden) if the current user does not have access to the example solution.
+     */
+    @GetMapping("/exercises/{exerciseId}/example-solution")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Exercise> getExerciseForExampleSolution(@PathVariable Long exerciseId) {
```

Comment:
Isn't this more or less a normal rest call to retrieve the exercise with an additional check if the example solution is already published?
If this includes the example solution as an attribute in the exercise you might have to check that it is not also included in many other rest calls returning an exercise

Reply:
This endpoint is intended to fetch the most basic exercise info without any categories, team config, posts, etc. to only get enough data to fill the screen below. I think having a separate endpoint for this has 2 advantages:

- Performance: Number of joins and extra queries are reduced.
- Simplicity & Security: Existing `exercises/{exerciseId}` and `/exercises/:exerciseId/details` are implemented with the assumption of students are not able to access the exam exercises at all, so I thought instead of relaxing the access rules there and making sure we do not leak any sensitive data, a new endpoint that just fetches the required data would be safer and makes the intent clearer.

![image](https://user-images.githubusercontent.com/18427209/203049514-9a66e8d6-0e30-46db-854d-7f52069ec082.png)

> If this includes the example solution as an attribute in the exercise you might have to check that it is not also included in many other rest calls returning an exercise

Overrides of `filterSensitiveInformation` at TextExercise, ModelingExercise etc. remove the example solution if it is not ready to be published yet from other endpoints that the students are allowed to access.